lib/if.c

```diff
@@ -231,6 +231,43 @@ struct interface *if_create_ifindex(ifindex_t ifindex, vrf_id_t vrf_id)
 	return ifp;
 }
 
+/* Create new interface structure. */
+extern void if_update_to_new_name(struct interface *ifp, const char *name)
+{
+	struct vrf *vrf = vrf_get(ifp->vrf_id, NULL);
+
+	assert(vrf);
+	if_set_name(ifp, name);
+
+	/*
+	 * HACK: Change the interface name in the running configuration
+	 * directly bypassing the northbound layer. This is necessary to
+	 * avoid deleting the interface and readding it with new name, which
+	 * would have several implications.
+	 */
+	if (yang_module_find("frr-interface")) {
+		struct lyd_node *if_dnode;
+		char oldpath[XPATH_MAXLEN];
+		char newpath[XPATH_MAXLEN];
+
+		snprintf(oldpath, sizeof(oldpath),
+			 "/frr-interface:lib/interface[name='%s'][vrf='%s']",
+			 ifp->name, vrf->name);
+		snprintf(newpath, sizeof(newpath),
+			 "/frr-interface:lib/interface[name='%s'][vrf='%s']",
+			 name, vrf->name);
+		if_dnode = yang_dnode_getf(running_config->dnode, "%s/vrf",
```

Comment:
You're not changing the name here, you're changing the VRF. More than that, oldpath and newpath are the same because you're using `ifp->name` and `name` to distinguish and they are the same after the call to `if_set_name` at the beginning of the function. Is this code tested?

Reply:
I tested it like below:
1. vtysh
```
config terminal
password zebra
interface dum3
ip address 4.4.4.4/24
end
```
2. check yang context: ( telnet 0 2601, then enable)
```
#show yang operational-context /frr-interface:lib
```
3. shell creation
```
ip link add dum3 type dummy
ip link set dev dum3 up
```
4. interface rename
```
ip link set dev dum3 down
ip link set dev dum3 name dum4
```
5. check: show yang operational
```
dut-vm# show yang operational-data /frr-interface:lib
{
  "frr-interface:lib": {
    "interface": [
      { 
        "name": "dum4",
        "vrf": "default",
        "state": {
          "if-index": 6,
          "mtu": 1500,
          "mtu6": 1500,
          "speed": 0,
          "metric": 0,
          "phy-address": "5e:37:28:23:c0:d6"
        },
        "frr-zebra:zebra": {
          "state": {
            "up-count": 1,
            "down-count": 1,
            "ptm-status": "disabled"
          }
        }
      },
```

I agree with the code done, there seems to be something wrong. however, I can see the operation context ok.
is there something else I should use ?

Reply:
Operational context is just a representation of the daemon's internal state.
You should use `show conf run json zebra` (or other daemon) to check the daemon's NB config.

Reply:
updated with appropriate code ( modifying interface name instead of vrf name).
and checked that with traces I enter in the change_leaf code.